streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -466,6 +510,11 @@ public void closeAndRecycleState() {
                 stateMgr.recycle();
                 recordCollector.close();
                 break;
+
+            case CLOSED:
+                log.trace("Skip close since state is {}", state());
```

Comment:
I think this might be one of those exceptions where we should still enforce that the state is not `CLOSED` (ie throw `IllegalStateException`) since there are related actions that occur outside of the Task implementation that will fail if we try to recycle a CLOSED task. Similar to prepare/post commit, resume, etc